editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/maven/CamelProjectConfigurator.java

```diff
@@ -68,7 +68,9 @@ public void mavenProjectChanged(MavenProjectChangedEvent event, IProgressMonitor
 			IProject project = facade.getProject();
 			MavenProject mavenProject = facade.getMavenProject(monitor);
 			IFacetedProject fproj = ProjectFacetsManager.create(project);
-			installDefaultFacets(project, mavenProject, fproj, monitor);
+			if (checkCamelContextsExist(project)) {
```

Comment:
I just realized that it means that we will slow down every Maven import because we will wait for a complete build to have finished in order to search for Java possible file extending RouteBuilder.
I have no solution yet... (and it will be always better than a NPE and isntalling Camel facets on a project which is not related^^)
But we need to start thinking about it